src/ThreeEditor/util/Detect/DetectTypes.ts

```diff
@@ -0,0 +1,73 @@
+export type Mesh = {
+    width: number,
+    height: number,
+    depth: number,
+    widthSegments: number,
+    heightSegments: number,
+    depthSegments: number,
+};
+
+export type Cyl = {
+    radius: number,
+    height: number,
+    radialSegments: number,
+    heightSegments: number
+};
+
+export type Zone = {
+    zoneId: number
+};
+
+export type All = {};
+
+export type Any = {
```

Comment:
https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#intersection-types